src/test/java/io/weaviate/integration/client/schema/ClientSchemaTenantOffloadingTest.java

```diff
@@ -0,0 +1,141 @@
+package io.weaviate.integration.client.schema;
+
+import io.weaviate.client.Config;
+import io.weaviate.client.WeaviateClient;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.v1.data.model.WeaviateObject;
+import io.weaviate.client.v1.schema.model.ActivityStatus;
+import io.weaviate.client.v1.schema.model.Tenant;
+import io.weaviate.integration.client.WeaviateDockerCompose;
+import io.weaviate.integration.client.WeaviateTestGenerics;
+import io.weaviate.integration.client.WeaviateVersion;
+import java.util.Arrays;
+import java.util.List;
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+public class ClientSchemaTenantOffloadingTest {
+
+  private WeaviateClient client;
+  private WeaviateTestGenerics testGenerics;
+
+  @ClassRule
+  public static WeaviateDockerCompose compose = new WeaviateDockerCompose(WeaviateVersion.WEAVIATE_IMAGE, true);
+
+  @Before
+  public void before() {
+    String httpHost = compose.getHttpHostAddress();
+    String grpcHost = compose.getGrpcHostAddress();
+    Config config = new Config("http", httpHost);
+    config.setGRPCSecured(false);
+    config.setGRPCHost(grpcHost);
+
+    client = new WeaviateClient(config);
+    testGenerics = new WeaviateTestGenerics();
+  }
+
+  @After
+  public void after() {
+    testGenerics.cleanupWeaviate(client);
+  }
+
+  @Test
+  public void shouldOffloadTenants() throws InterruptedException {
+    // create tenants and class
+    String className = "Pizza";
+    String[] tenants = new String[]{"Tenant1", "Tenant2", "Tenant3"};
+    Tenant[] tenantObjs = Arrays.stream(tenants)
+      .map(tenant -> Tenant.builder().name(tenant).build())
+      .toArray(Tenant[]::new);
+    testGenerics.createSchemaPizzaForTenants(client);
+    testGenerics.createTenantsPizza(client, tenantObjs);
+    // verify tenants existence
+    Result<List<Tenant>> getResult = client.schema().tenantsGetter()
+      .withClassName(className)
+      .run();
+    assertThat(getResult).isNotNull()
+      .returns(false, Result::hasErrors)
+      .extracting(Result::getResult).asList()
+      .hasSize(tenants.length);
+    // insert data
+    testGenerics.createDataPizzaForTenants(client, tenants);
+    // verify data existence
+    for (String tenant : tenants) {
+      Result<List<WeaviateObject>> result = client.data().objectsGetter().withClassName(className).withTenant(tenant).run();
+      assertThat(result).isNotNull()
+        .returns(false, Result::hasErrors)
+        .extracting(Result::getResult).asList().hasSize(4);
+    }
+    // verify tenant status HOT
+    verifyEventuallyTenantStatus(className, ActivityStatus.HOT);
+    // update tenants to FROZEN
+    updateTenantStatus(className, tenants, ActivityStatus.FROZEN);
+    // verify tenant status FREEZING
+    verifyEventuallyTenantStatus(className, ActivityStatus.FREEZING);
```

Comment:
IMO this might introduce flakiness. As I understand `FREEZING` is intermediate status before tenant is finally set to `FROZEN`. If offloading is fast test might not catch tenant in `FREEZING` state

Reply:
I have copied the test case that is in Weaviate and there is this check, I can change this to `FROZEN`, wdyt?